src/Offcanvas.php

```diff
@@ -4,235 +4,544 @@
+     */
+    public function responsive(Responsive|string $size): self
     {
-        parent::begin();
+        $size = $size instanceof Responsive ? $size->value : $size;
 
-        $options = $this->options;
-        $bodyOptions = $this->bodyOptions;
-        $tag = ArrayHelper::remove($options, 'tag', 'div');
-        $bodyTag = ArrayHelper::remove($bodyOptions, 'tag', 'div');
+        $new = clone $this;
+        $new->responsive = match ($size) {
+            'sm', 'md', 'lg', 'xl', 'xxl' => $size,
+            default => '',
+        };
 
-        Html::addCssClass($options, ['widget' => 'offcanvas', 'placement' => $this->placement]);
-        Html::addCssClass($bodyOptions, ['widget' => 'offcanvas-body']);
+        return $new;
+    }
 
-        $options['id'] = $this->getId();
-        $options['tabindex'] = -1;
+    /**
+     * Sets whether the offcanvas component is scrollable.
+     *
+     * @return self A new instance with the specified scrollable setting.
+     */
+    public function scrollable(): self
+    {
+        $new = clone $this;
+        $new->scrollable = true;
 
-        if ($this->title !== null && $this->title !== '') {
-            if (isset($this->titleOptions['id'])) {
-                $options['aria-labelledby'] = $this->titleOptions['id'];
-            } elseif ($options['id']) {
-                $options['aria-labelledby'] = $options['id'] . '-title';
-            }
-        }
+        return $new;
+    }
 
-        if ($this->scroll) {
-            $options['data-bs-scroll'] = 'true';
-        }
+    /**
+     * Sets whether the offcanvas component is visible.
+     *
+     * @param bool $show Whether the offcanvas component is visible.
+     *
+     * @return self A new instance with the specified visibility setting.
+     */
+    public function show(): self
+    {
+        $new = clone $this;
+        $new->show = true;
 
-        if ($this->withoutBackdrop) {
-            $options['data-bs-backdrop'] = 'false';
-        }
+        return $new;
+    }
+
+    /**
+     * Sets the theme for the offcavas component.
+     *
+     * @param string $theme The theme for the offcanvas component.
+     *
+     * @return self A new instance with the specified theme.
+     */
+    public function theme(string $theme): self
+    {
+        return $this->addAttributes(['data-bs-theme' => $theme === '' ? null : $theme]);
+    }
 
-        if ($this->theme) {
-            $options['data-bs-theme'] = $this->theme;
+    /**
+     * Sets the title of the offcanvas component.
+     *
+     * @param string|Stringable $title The title of the offcanvas component.
+     *
+     * @return self A new instance with the specified title.
+     */
+    public function title(string|Stringable $title): self
+    {
+        $new = clone $this;
+        $new->title = $title;
+
+        return $new;
+    }
+
+    /**
+     * Sets the HTML attributes for the offcanvas title component.
```

Comment:
```suggestion
     * Sets the HTML attributes for the title.
```

Reply:
Let's try to make descriptions shorter but cleaner. In this and many other cases we already know that it's offcanvas so we don't need to repeat it.

Reply:
done.